src/server/admin/decommission-service/controllers/finish.js

```diff
@@ -0,0 +1,64 @@
+import Joi from 'joi'
+import Boom from '@hapi/boom'
+
+import { sessionNames } from '~/src/server/common/constants/session-names.js'
+import { deleteDeploymentFiles } from '~/src/server/admin/decommission-service/helpers/fetch/delete-deployment-files.js'
+import { deleteEcs } from '~/src/server/admin/decommission-service/helpers/fetch/delete-ecs.js'
+import { isServiceValid } from '~/src/server/admin/decommission-service/helpers/is-service-valid.js'
+
+export const decommissionFinishController = {
+  options: {
+    validate: {
+      params: Joi.object({
+        serviceName: Joi.string().required()
+      }),
+      failAction: () => Boom.boomify(Boom.notFound())
+    }
+  },
+  handler: async (request, h) => {
+    const serviceName = request.params.serviceName
+    const authedUser = await request.getUserSession()
+
+    const serviceIsValid = await isServiceValid(serviceName, request)
+
+    if (!serviceIsValid) {
+      return h.redirect(`/admin/decommission-service/${serviceName}/step-2`)
+    }
+
+    try {
+      const deleteFilesResponse = await deleteDeploymentFiles(
+        request,
+        serviceName
+      )
+      const deleteEcsResponse = await deleteEcs(request, serviceName)
+      if (deleteFilesResponse?.ok && deleteEcsResponse?.ok) {
+        request.yar.clear(sessionNames.validationFailure)
+        await request.yar.commit(h)
+
+        request.yar.flash(sessionNames.notifications, {
+          text: 'Service decommissioned successfully',
+          type: 'success'
+        })
+
+        request.audit.sendMessage({
+          event: `Service decommissioned: ${serviceName} by ${authedUser.id}:${authedUser.displayName}`,
+          data: {
+            serviceName
+          },
+          user: request.pre.authedUser
```

Comment:
you don't have a the `pre` setup. So this won't ever have a value, it will always be `undefined`. You need to add:

```
options: {
    pre: [provideAuthedUser],
    validate: {
      params: Joi.object({
        serviceName: Joi.string().required()
      }),
      failAction: () => Boom.boomify(Boom.notFound())
    }
  }
```